pkg/multicluster/reconcile_loop.go

```diff
@@ -0,0 +1,94 @@
+package multicluster
+
+import (
+	"context"
+
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/skv2/pkg/ezkube"
+	"github.com/solo-io/skv2/pkg/reconcile"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type Reconciler interface {
+	// reconcile an object
+	// requeue the object if returning an error, or a non-zero "requeue-after" duration
+	Reconcile(cluster string, object ezkube.Object) (reconcile.Result, error)
+}
+
+type DeletionReconciler interface {
+	// we received a reconcile request for an object that was removed from the cache
+	ReconcileDeletion(cluster string, request reconcile.Request)
+}
+
+// a Reconcile Loop runs resource reconcilers until the context gets cancelled
+type Loop interface {
+	// RunReconciler adds a reconciler to a slice of reconcilers that will be run against
+	RunReconciler(ctx context.Context, reconciler Reconciler, predicates ...predicate.Predicate) error
```

Comment:
I think I'll rename this since it doesn't really "Run" anything in this case, just add to a slice. Something like "RegisterReconciler", open to suggestions.